### DISCUSSION TITLE
A question about loading auxiliary images

### DISCUSSION THREAD

**Discussion by EdyQiu** [open-mmlab/mmclassification#1391] [General] [answered]
Hi, I want to implement an idea that when loading each image in the dataset,  its edge-extracted version is also loaded. 
If I simply change the annotation file, it seems random sampling would stop me from achieving the goal. 
I also tried to modify the train-pipeline, but still feel hard to find a good way to achieve my goal.

Any help about how could I modify the code would be very helpful!

**Comment by Ezra-Yu** [comment, marked as answer]
>  its edge-extracted version is also loaded.
Do you mean using the edge-extracted to do classification?

1. Use `YourDataset` without `LoadImageFromFile` transforms refer to the [tutorial](https://mmclassification.readthedocs.io/en/dev-1.x/advanced_guides/datasets.html#create-dataset-class). return `{'img': img, 'gt_label': gt_label}` as https://github.com/open-mmlab/mmclassification/blob/fb16bdc6a25f96843f8fb97b02e45872c61dc566/mmcls/datasets/cifar.py#L91-L94
2. Add a new Transform to get the `edge-extracted version`.

**Reply by EdyQiu** [reply]
Thanks for your reply!

The edge-extracted version is not used for classification directly, I want to use it to enhance the edge information of the original image, and use the mixed image for classification. (Which means the label of the edge-extracted version does not matter)

I think you are suggesting me to write a new transformation to replace "LoadImageFromFile" transformation in the training pipeline right? meanwhile, I also need to write a new dataset class, or modifying the "load annotation" function of the original dataset class.

**Reply by Ezra-Yu** [reply]
Yes, If your dataset is very small as minist and cifra, I think you can just implement a new Dataset class to replace the CustomDataset and 'LoadImageFromFile'. 
If your dataset is very big. modify the CustomDataset  to save the edge-extracted images info and add a "LoadMixedImageFromFile" transformation.
